### ISSUE TITLE
The certification key values do not get updated on app startup or sync button press.

### ISSUE THREAD

**Issue by eetupa** [eu-digital-green-certificates/dgca-verifier-app-android#163]
## Describe the bug

The certification key values do not get updated on app startup or sync button press. 

**TLDR**: 
This is caused by storing the resumeToken in VerifierRepositoryImpl. The first time the app is started all cert key values are downloaded and stored fine in a chain of requests and the last resumeToken of the request chain is stored in preferences. This causes the app to start the next cert key value update request chain with the last resumeToken received from the previous request chain. The request with the last resumeToken gives a HTTP 204 response which causes the request chain to stop right away. The same resumeToken is left stored and the next time the app starts or the sync button is pressed the same thing happens again. Which means the cert key values are never updated.

**A longer version** where I tried to figure out and write down how the app works in this regard:

**When starting the app for the first time:**
- The cert-key-ids are requested from the /signercertificateStatus endpoint.
- Then the cert-key-values are requested from the /signercertificateUpdate endpoint and the retrieved values are stored in the db.
    - This is done by adding a resumeToken header to the cert-key-value request. The resumeToken tells the backend which cert-key-value we want to get. **(1)**
    - The first cert-key-value request is done with a resumeToken fetched from the app preferences. Empty string "" is used if no resumeToken is found, which is true when starting the app for the very first time.
    - The cert-key-value request response contains the resumeToken for the next request in its headers. This new resumeToken is stored in preferences. **(2)**
    - A new cert-key-value request is made with the stored resumeToken again and again until the backend responds with a HTTP 204 (no content)
    - This happens when there are no cert-key-value available for the resumeToken. After the HTTP 204 response no more requests are made.
    - The resumeToken that returns HTTP 204 from the backend is kept stored in preferences.

**When the app is started for the second time or the sync button is pressed:**
- The above process starts again.
- Only this time there is a value present in preferences.resumeToken. And that value is not empty, but the last resumeToken of the above process.
- This means that only one cert-key-value request is made and this request responds with HTTP 204, which means the above process stops, which means the cert-keys are not updated. **(3)**

## Expected behaviour

When starting the app or when pressing the sync button I expect the certification key values to be fetched from the backend and stored in the app db.

## Steps to reproduce the issue

Perform a new install on a device without the app. Start the app for the first time and monitor the HTTP calls in the log (especially the amount of HTTP calls made). Then press the sync button or close and start the app again and monitor the logs again.

## Possible Fix

These numbers refer to the longer version of the bug description.

**(1)** I feel like this could be simpler. We could make the request for the cert-key-value with the cert-key-id instead of using resumeTokens. Or better yet make a bulk endpoint that takes in an array of cert-key-ids and returns the cert-key-values for the requested cert-key-ids. Or we could simply have an endpoint that returns all the cert-keys in a list. These last two suggestions would greatly decrease the amount of requests made.
**(2)** The preferences.resumeToken is not used anywhere else so I don't think there's a reason for storing it. In my opinion it could be removed from the preferences, unless starting the request chain where it was left the last time is actually the desired functionality.
**(3)** This causes no new cert-key-values to be fetched after the app is ran for the first time. This must be a bug or I've misunderstood something. This could be fixed by doing (2) which would cause the app to call the backend with an empty resumeToken, which would result in over a hundred times every time it is started or synced. Or it could be fixed by doing (1), which is a lot of work because it requires changes to the backend endpoints. Or it could be fixed by some other more sophisticated way (eg. comparing some sort of delta of which certs have changed and fetch only those) which would also require more endpoints to the backend.

## Additional context

Our app was deployed to production three weeks ago. Last week we noticed that there were lots of unsuccessful scans coming from EL (Greece), HR (Croatia), IE (Ireland) and IS (Iceland). Turned out that our app had not received the certificate keys for these countries due to a http request throttling issue. This issue has been resolved but our problem persists. We are unable to fetch the missing certificate keys without uninstalling or clearing the data of the installed apps, which unfortunately in our case is not possible. This led me to investigate how the app actually updates the certificate keys.

**Comment by oleksandrsarapulovgl** [comment]
PR on the client side: https://github.com/eu-digital-green-certificates/dgca-verifier-app-android/pull/177
